src/helpers/pinConnections.ts

```diff
@@ -0,0 +1,114 @@
+ * limitations under the License.
+ ***************************************************************************** */
+
+import { IPosition } from 'monaco-editor';
+import { IPinConnections } from '../components/links/BoxConnections';
+import { ExtendedConnectionOwner, Pin } from '../models/Box';
+import { Link } from '../models/LinksDefinition';
+import { BoxesStore } from '../stores/BoxesStore';
+import { BoxUpdater } from '../stores/BoxUpdater';
+
+export type PinsPositions = {
+	connections: number;
+	position: IPosition;
+};
+
+export type PinsInfo = {
+	name: string;
+	numOfConnections: number;
+};
+
+export type InvalidLink = {
+	link: Link<ExtendedConnectionOwner>;
+	lostBoxes: {
+		box: string;
+	}[];
+	lostPins: {
+		pin: string;
+		box: string;
+	}[];
+};
+
+export function getCountPinsConnections(
+	boxesStore: BoxesStore,
+	boxUpdater: BoxUpdater,
+): PinsInfo[] | null {
+	const selectedBoxPins = boxesStore.selectedBox?.spec.pins;
+	const [toConnections, fromConnections] = boxUpdater.selectedBoxConnections;
+	const allConnections = [...(toConnections?.pins || []), ...(fromConnections?.pins || [])];
+	if (selectedBoxPins && allConnections) {
+		const pinsInfoArray: PinsInfo[] = [];
+		selectedBoxPins.forEach((pin: Pin, index: number) => {
+			pinsInfoArray[index] = { name: pin.name, numOfConnections: 0 };
+			allConnections.forEach((iPinConnection: IPinConnections) => {
+				if (JSON.stringify(iPinConnection.pin) === JSON.stringify(pin)) {
+					pinsInfoArray[index] = {
+						name: pin.name,
+						numOfConnections: pinsInfoArray[index].numOfConnections + iPinConnection.boxes.length,
+					};
+				}
+			});
+		});
+		return pinsInfoArray;
+	}
+	return null;
+}
+
+export function detectInvalidLinks(boxesStore: BoxesStore, boxUpdater: BoxUpdater): InvalidLink[] {
+	const invalidLinks: InvalidLink[] = [];
+	boxUpdater.links.forEach(link => {
+		var invalidLink: InvalidLink = {
+			link: link,
+			lostBoxes: [],
+			lostPins: [],
+		};
+		for (let i = 0; i < 2; i++) {
+			const box = boxesStore.boxes.find(
+				box => box.name === (i === 0 ? link.from?.box : link.to?.box),
+			);
+			if (box === undefined) {
+				invalidLink.lostBoxes.push({
+					box: i === 0 ? link.from?.box || '' : link.to?.box || '',
+				});
+			} else {
+				const pins = box.spec.pins?.find(
+					pin => pin.name === (i === 0 ? link.from?.pin : link.to?.pin),
+				);
+				if (pins === undefined) {
+					invalidLink.lostPins.push({
+						pin: i === 0 ? link.from?.pin || '' : link.to?.pin || '',
+						box: box.name,
+					});
+				}
+			}
+		}
+		if (invalidLink.lostBoxes.length > 0 || invalidLink.lostPins.length > 0) {
+			invalidLinks.push(invalidLink);
+		}
+	});
+	return invalidLinks;
+}
+
+export function deleteInvalidLinks(invalidLinks: InvalidLink[], boxUpdater: BoxUpdater) {
+	invalidLinks.forEach(link => {
+		boxUpdater.deleteLink(link.link);
+	});
+}
+
+export function selectBox(boxName: string, boxesStore: BoxesStore) {

```

Comment:
I think we can extend `selectBox` method inside boxesStore by passing `string` param to it instead of this method